umbraco-deploy/troubleshooting.md

```diff
@@ -10,3 +10,67 @@ If you are unable to find the issue you are having, then please reach out to our
+
+Umbraco Deploy maintains a cached set of signatures that represent each schema and content item. They are used when transferring or restoring content between environments to aid performance.
+
+If having resolved schema mismatches you still have reports of errors, it might be that the signatures are out of date.  In other words, Deploy is using a cached representation of an item that no longer matches the actual item stored in Umbraco.
+
+This should not be necessary in normal use, but can occur after upgrades. If you have this situation, you can clear the cached signatures in both the upstream and downstream environments.  You do this via the *Clear Cached Signatures* operation available on the _Settings > Deploy_ dashboard:
+
+![Clear cached signatures](images/clear-cached-sigs.png)
+
+## Slow responses or timeouts when restoring or transferring
+
+When transferring or restoring content between environments, Deploy needs to ensure that all related items are updated together.  It also checks that any schema dependencies an item has also exist in the target environment. When a large amount of content is selected for transfer or restore, this process of determining all the dependent items can take some time.
+
+If you find the process slow or timing out, there are few options you can take.
+
+### Review timeouts
+
+Firstly, you can review and update the [timeout settings available with Deploy](./deploy-settings.md#timeout-settings). Increasing these from the default values may help, but won't necessarily resolve all issues. This is because some timeouts are fixed values set by the hosting environment.
+
+### Use batch configuration for transfers to upstream environments
+
+If transferring items from a downstream environment to an upstream one, it's possible to [configure a batch size](./deploy-setings#batch-settings).  With this in place, transfers will be batched into separate operations, allowing each single operation to complete before any hosting environment enforced timeout.
+
+### Ensure signatures are pre-cached
+
+For transfer or restore operations, it's worth ensuring Deploy's cached signatures are fully populated in both the upstream and downstream environments.  This can be done via the *Set Cached Signatures* operation available on the _Settings > Deploy_ dashboard:
+
+![Set cached signatures](images/set-cached-sigs.png)
+
+Now the checks Deploy has to do to figure out the items and dependencies to process will complete much more quickly.
+
+### Modify the checksum calculation method for media files
+
+Deploy will do comparisons between the entities in different environments to determine if they match and decide whether to include them in the operation. By default, for media files, a check is made on the a portion of the intial bytes of the file.
```

Comment:
This sentence: _By default, for media files, a check is made on **the a portion of the intial bytes of the file**._ is the same as my comment above, is it on purpose with the a byte or is it just an error? 🤔